src/language_features/document_symbol.rs

```diff
@@ -216,14 +235,41 @@
     ctx: &Context,
 ) -> String {
     fn format_symbol_at_depth<'a, T: Symbol<T>>(
-        output: &mut Vec<(String, String, &'a str)>,
+        output: &mut Vec<(String, String, String)>,
         items: &'a [T],
         meta: &EditorMeta,
         server: &ServerSettings,
         ctx: &Context,
         depth: usize,
+        prefix: &Vec<bool>,
     ) {
-        for symbol in items {
+        let length = items.len();
+        let is_root = depth == 0;
+        for (index, symbol) in items.iter().enumerate() {
+            let is_last = index + 1 == length;
+
+            let mut new_prefix = prefix.clone();
+            new_prefix.push(is_last);
```

Comment:
Should not need to clone the entire Vec here, instead push to the existing one and pop it at the end of the loop iteration

Reply:
Sorry, I didn't understand what you meant.

The algorithm build a prefix string of booleans as _"is this symbol at the depth level (marked by a vector index) is a last child of its parent?"_. For some symbols it will be `111`, for some `010`, etc. It's probably possible, but I'm not sure how can I reuse the same vector.

Reply:
we can reuse it because we're doing a [depth-first traversal](https://en.wikipedia.org/wiki/Depth-first_search).
The prefix vector is a stack of all ancestors of the current node, which is exactly what we need to compute the prefixed pipes.

Reply:
Yeah, [I see now](https://github.com/kak-lsp/kak-lsp/commit/4dd8e2a2a0222827a43808be9ed3c7483a6b1ffb#diff-aea8cfa1bdea7837f8efd39fe43fd6a66d98a9724b536736b10a97206d98478dR289-R291), cool!